packages/engine/core/src/lib/engine.ts

```diff
@@ -136,6 +136,50 @@ export class Engine {
     return this.plugins[target]['addRequest'](request, method, payload)
   }
 
+  /**
+   * Cancels calls from a plugin to another
+   * @param caller The name of the plugin that calls the method
+   * @param path The path of the plugin that manages the method
+   * @param method The name of the method to be cancelled, if is empty cancels all calls from plugin
+   */
+  private async cancelMethod(caller: string, path: string, method: string) {
+    const target = path.split('.').shift()
+    if (!this.plugins[target]) {
+      throw new Error(`Cannot call ${target} from ${caller}, because ${target} is not registered`)
+    }
+
+    // Get latest version of the profiles
+    const [to, from] = await Promise.all([
+      this.manager.getProfile(target),
+      this.manager.getProfile(caller),
+    ])
+
+    // Check if plugin FROM can activate plugin TO
+    const isActive = await this.manager.isActive(target)
+    
+    if (!isActive) {
```

Comment:
if `target` is not active you probably don't need to do anything..
at least `await this.manager.toggleActive(target)` doesn't seem needed